src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -195,14 +195,67 @@ public static string LookupTargetFrameworkFromProjectFile(string projectLocation
             return element?.Value;
         }
 
-        public static string LookupOutputTypeFromProjectFile(string projectLocation)
+        /// <summary>
+        /// Uses `dotnet msbuild` to determine project properties.
+        /// </summary>
+        /// <param name="projectLocation">Path of the project</param>
+        /// <param name="propertyName">The name of the property</param>
+        /// <returns>The value of the project property</returns>
+        private static string LookupMsbuildProperty(string projectLocation, string propertyName)
         {
-            var projectFile = FindProjectFileInDirectory(projectLocation);
```

Comment:
The `dotnet msbuild -getProperty` feature is only available for .NET 8+. We could have users especially in CI systems that are only running with .NET 6. You need to wrap this in a compilation conditional and fallback to the old system when not using .NET 8.

Another option is to always try this approach but if it fails then fallback to the old approach. That would allow us to use this technic inside Visual Studio. But you need to make sure we detect the error correctly.

Reply:
I went with the 2nd approach. I will always try to use `dotnet msbuild -getProperty`, but if you are using a .NET SDK prior to NET8, that command will throw the following error: `MSBUILD : error MSB1001: Unknown switch.`. This is thrown when you use: `-getProperty`. In .NET8, if you use that switch with a property type that doesn't exist, the CLI returns empty. It doesn't throw this error.

Reply:
Possible alternative approach using `Microsoft.Build` and `Micorosft.Build.Locator`.

I had looked at this for Lambda Annotations, but couldn't use that there because source generators have to target .NET Standard and these packages don't. 

I haven't explored all the tradeoffs, but might be nice to not have to manage our own `Process`. There are some implications around Visual Studio compatibility when selecting versions of these dependencies.

```
using Microsoft.Build.Evaluation;
using Microsoft.Build.Locator;

var instance = MSBuildLocator.QueryVisualStudioInstances().OrderByDescending(
       instance => instance.Version).First(); // select which of the installed instances to use

// Register a specific instance of MSBuild
MSBuildLocator.RegisterInstance(instance);

DetectProjectType("path\\to\\project.csproj");

static void DetectProjectType(string pathToCsproj)
{
    var project = new Project();

    var outputType = project.GetProperty("OutputType");

    Console.WriteLine(outputType.EvaluatedValue);
}
```